metafix/src/main/java/org/metafacture/metafix/FixPath.java

```diff
@@ -276,7 +279,15 @@ else if (isReference(field)) {
         }
         else {
             if (!hash.containsField(field)) {
-                hash.put(field, Value.newHash().withPathSet(newValue.getPath()));
+                final Value value = Arrays.asList(ReservedField.$prepend.name(), ReservedField.$append.name())
+                        .contains(tail(path)[0]) ? Value.newArray() : Value.newHash();
+                hash.put(field, value.withPathSet(newValue.getPath()));
+            }
+            else {
+                final Value value = hash.get(field);
+                if (value.isString()) {
+                    hash.put(field, Value.newArray(a -> a.add(value)));
+                }
             }
             insertInto(hash.get(field), mode, newValue, field, tail(path));
```

Comment:
There might an opportunity here to save the additional `hash.get()` and `tail()` calls.

Reply:
Yes, thanks, refactored to avoid both.